### ISSUE TITLE
Allow plotting weight in Graph.plot

### ISSUE THREAD

**Issue by martinfleis** [pysal/libpysal#628]
A side note - I also realised that we can directly plot weight. Neat.
![Screenshot 2023-11-03 at 23 08 10](https://github.com/pysal/libpysal/assets/36797143/8f1ce4e2-72b5-42d2-a073-a5a7bcb311c5)

_Originally posted by @martinfleis in https://github.com/pysal/libpysal/issues/627#issuecomment-1793159245_
            

In explore, we get ability to plot the actual weight on a map for free but with static plots, we need to get matplotlib colormap, retrieve the colours and pass them to the `LineCollection` ourselves. Might be worth exploring if there's an appetite.

**Comment by jGaboardi** [comment]
xref #562

**Comment by jGaboardi** [comment]
@martinfleis This can now be closed out, correct?

**Comment by knaaptime** [comment]
once you're [here](https://github.com/pysal/libpysal/blob/8c4a2447685ebc9ec10d19cd04844dd04e862bc1/libpysal/graph/_plotting.py#L111), you have an array of edges keyed by id? if you could bring along the weight values by going to a dataframe then joining or something, you could grab an [array of colors](https://pysal.org/mapclassify/generated/mapclassify.util.get_color_array.html#mapclassify.util.get_color_array) in the proper alignment and send that?

**Comment by knaaptime** [comment]
yeah, though could also let the user choose whether to plot incoming values or outgoing values  with a kwarg then keep j-i or i-j and join weight values accordingly
